### DISCUSSION TITLE
CRSF bad request error when redirected from another website

### DISCUSSION THREAD

**Discussion by furkan-pensa** [apache/airflow#36950] [General]
### Apache Airflow version

2.2.5

### What happened

We are giving a link to the Airflow webserver running in a docker container from our UI and when the Airflow UI is opened via that link it errors out saying "Bad Request. The CSRF session token is missing.". When this happens there is no reliable way to make the webserver work again. When I type the URL and and access the webserver, it works until I click on the link on our UI. Then it breaks and doesn't work even if I type the URL.

### What you think should happen instead

There should be a more clear documentation on how to configure and use the CSRF token (aka `secret_key` in webserver section of airflow.cfg).

### How to reproduce

From another website access the Airflow webserver using a hyperlink like below.
```html
<a
  :href="`http://localhost:8080`"
  target="_blank"
>
  Airflow
</a>
```

### Operating System

Ubuntu

### Versions of Apache Airflow Providers

apache/airflow:2.2.5-python3.9 docker image

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by Tejalgnrml** [comment]
Similar issue, upon redirecting the webserver ip through a domain i am facing this issue
Any solutions?
Thanks in advance

**Comment by potiuk** [comment]
Possibly the guide might help you. https://airflow.apache.org/docs/apache-airflow/stable/howto/run-behind-proxy.html

**Comment by potiuk** [comment]
Converting it into discussion. Many of our users follow our docs and generic way of setting up Airflow behind proxies - similar to other apps they do, and the docs we have seem to be already good-enought for that. If anyone has a different case there and provides more information, the ways they tried and struggled with and eventually succceded, they are probably the best ones to propose a PR in our docs to suggest "better documentation" of those cases. Airflow documentation (including our HowTo guides) have been contributed by many users (we have almost 2800 of them) and contributions to How To guides from such users are the best as they are describing the problems and solutions from their perspectives.
